src/executor/attach_op_execs_pass.cc

```diff
@@ -201,13 +201,45 @@ class FComputeExecutor : public StorageFallbackOpExecutor {
 
 // fcompute_ex executor
 class FComputeExExecutor : public OpExecutor {
+#if MXNET_USE_MKLDNN == 1
+  // Does any of the arrays use MKLDNN format.
+  static bool IsMKLDNNData(const std::vector<NDArray> &arrs) {
+    for (auto &arr : arrs)
+      if (arr.IsMKLDNNData())
+        return true;
+    return false;
+  }
+#endif
  public:
   void Run(RunContext rctx, bool is_gpu) override {
     op_ctx.run_ctx = rctx;
+    std::vector<NDArray> *inputs = &in_array;
 #if MXNET_USE_MKLDNN == 1
+    std::vector<NDArray> in_bufs;
     InvalidateOutputs(out_array, req);
+    // If this operator isn't an MKLDNN operator, we need to manually convert
+    // the layout before calling FComputeEx. This is kind of similar to
+    // fallback. However it only handles data with the MKLDNN format.
+    // There is a potential race condition if we don't convert the layout in
+    // advance. It happens if an NDArray from an MKLDNN operator is used by a
+    // non-MKLDNN operator that supports FComputeEx and takes NDArrays with the
+    // default storage type as inputs (This happens rarely).
```

Comment:
This comment sounds like the issue will not be resolved by this change. Am I misinterpreting it?